Marker/Classes/TextStyle+Extensions.swift

```diff
@@ -0,0 +1,173 @@
+        let strikethroughStyleToUse = (newStrikethroughStyle == nil) ? self.strikethroughStyle : newStrikethroughStyle!
+        let strikethroughColorToUse = (newStrikethroughColor == nil) ? self.strikethroughColor : newStrikethroughColor!
+        let underlineStyleToUse = (newUnderlineStyle == nil) ? self.underlineStyle : newUnderlineStyle!
+        let underlineColorToUse = (newUnderlineColor == nil) ? self.underlineColor : newUnderlineColor!
+        let textTransformToUse = (newTextTransform == nil) ? self.textTransform : newTextTransform!
+        
+        return TextStyle(
+            font: fontToUse,
+            emFont: emFontToUse,
+            strongFont: strongFontToUse,
+            textColor: textColorToUse,
+            characterSpacing: characterSpacingToUse,
+            lineSpacing: lineSpacingToUse,
+            lineHeightMultiple: lineHeightMultipleToUse,
+            minimumLineHeight: minimumLineHeightToUse,
+            maximumLineHeight: maximumLineHeightToUse,
+            firstLineHeadIndent: firstLineHeadIndentToUse,
+            headIndent: headIndentToUse,
+            paragraphSpacing: paragraphSpacingToUse,
+            paragraphSpacingBefore: paragraphSpacingBeforeToUse,
+            textAlignment: textAlignmentToUse,
+            lineBreakMode: lineBreakModeToUse,
+            strikethroughStyle: strikethroughStyleToUse,
+            strikethroughColor: strikethroughColorToUse,
+            underlineStyle: underlineStyleToUse,
+            underlineColor: underlineColorToUse,
+            textTransform: textTransformToUse
+        )
+    }
+    
+    /// Creates new TextStyle from exisiting TextStyle, updating fonts with new size.
+    ///
+    /// - Parameter newFontSize: New font size.
+    /// - Returns: New TextStyle.
+    public func with(newFontSize: CGFloat) -> TextStyle {
+        return self.with(
+            newFont: self.font.withSize(newFontSize),
+            newEmFont: self.emFont.withSize(newFontSize),
+            newStrongFont: self.strongFont.withSize(newFontSize)
+        )
+    }
+    
+    /// Creates new TextStyle from exisiting TextStyle, replacing font with bold variation.
+    ///
+    /// - Returns: New TextStyle with updated font.
+    public func bold() -> TextStyle {
+        return self.with(newFont: strongFont)
+    }
+    
+    /// Creates new TextStyle from exisiting TextStyle, replacing font with italic variation.
+    ///
+    /// - Returns: New TextStyle with updated font.
+    public func italic() -> TextStyle {
+        return self.with(newFont: emFont)
+    }
+    
+    /// Creates new Text Style from exisiting TextStyle, with underline.
+    ///
+    /// - Parameter color: Underline color. Defaults to textColor.
+    /// - Parameter style: Underline style. Defaults to single line.
+    /// - Returns: New TextStyle with underline.
+    public func underlined(color: UIColor? = nil, style: NSUnderlineStyle = .styleSingle) -> TextStyle {
```

Comment:
Not sure if we should create individual functions for `bold`, `italic` and `underlined`... It might lead us down to a rabbit hole that we might not necessarily wanna go down.